src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,120 @@
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '.';
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var position = context.Position;
+            var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            if (token.IsKind(SyntaxKind.DotToken))
+            {
+                var parent = token.Parent;
+                var expression = parent switch
+                {
+                    MemberAccessExpressionSyntax memberAccess => memberAccess.Expression,
+                    _ => null,
+                };
+                if (expression != null)
+                {
+                    var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
+                    var container = semanticModel.GetTypeInfo(expression, cancellationToken).Type;
+                    var allMembers = container.GetMembers();
+                    var allExplicitConversions = from m in allMembers.OfType<IMethodSymbol>()
+                                                 where
+                                                    m.IsConversion() && // MethodKind.Conversion
+                                                    m.Name == WellKnownMemberNames.ExplicitConversionName && // op_Explicit
+                                                    m.Parameters[0].Type == container // Convert from container type to other type
+                                                 select SymbolCompletionItem.CreateWithSymbolId(
+                                                     displayText: $"({m.ReturnType.ToNameDisplayString()})", // The type to convert to
+                                                     symbols: ImmutableList.Create(m),
+                                                     rules: CompletionItemRules.Default,
+                                                     contextPosition: position,
+                                                     properties: ImmutableDictionary<string, string>.Empty.Add("IsConversion", "true"));
+                    context.AddItems(allExplicitConversions);
+                }
+            }
+        }
+
+        internal override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, TextSpan completionListSpan, char? commitKey, bool disallowAddingImports, CancellationToken cancellationToken)
```

Comment:
The following lines are just prototyping and the same code as in the alternative approach in SymbolCompletionProvider.